docker/crossbuild/Dockerfile.kinetic-crossbuild

```diff
@@ -47,60 +47,72 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
     libyaml-cpp-dev \
     python-flask \
     python-serial \
-    qtbase5-dev \
-    ros-kinetic-angles \
-    ros-kinetic-automotive-platform-msgs \
-    ros-kinetic-camera-info-manager \
-    ros-kinetic-catkin \
-    ros-kinetic-cmake-modules \
-    ros-kinetic-cv-bridge \
-    ros-kinetic-diagnostic-msgs \
-    ros-kinetic-diagnostic-updater \
-    ros-kinetic-dynamic-reconfigure \
-    ros-kinetic-geometry-msgs \
-    ros-kinetic-gps-common \
-    ros-kinetic-grid-map-cv \
-    ros-kinetic-grid-map-msgs \
-    ros-kinetic-grid-map-ros \
-    ros-kinetic-image-geometry \
-    ros-kinetic-image-transport \
-    ros-kinetic-imu-filter-madgwick \
-    ros-kinetic-imu-tools \
-    ros-kinetic-jsk-recognition-msgs \
-    ros-kinetic-jsk-rviz-plugins \
-    ros-kinetic-message-filters \
-    ros-kinetic-message-generation \
-    ros-kinetic-message-runtime \
-    ros-kinetic-nav-msgs \
-    ros-kinetic-nlopt \
-    ros-kinetic-nmea-msgs \
-    ros-kinetic-nodelet \
-    ros-kinetic-pcl-conversions \
-    ros-kinetic-pcl-msgs \
-    ros-kinetic-pcl-ros \
-    ros-kinetic-pluginlib \
-    ros-kinetic-rosconsole \
-    ros-kinetic-roscpp \
-    ros-kinetic-roslaunch \
-    ros-kinetic-roslib \
-    ros-kinetic-roslint \
-    ros-kinetic-rospy \
-    ros-kinetic-rostest \
-    ros-kinetic-rosunit \
-    ros-kinetic-rqt-plot \
-    ros-kinetic-rviz \
-    ros-kinetic-sensor-msgs \
-    ros-kinetic-shape-msgs \
-    ros-kinetic-sound-play \
-    ros-kinetic-std-msgs \
-    ros-kinetic-std-srvs \
-    ros-kinetic-stereo-msgs \
-    ros-kinetic-tf \
-    ros-kinetic-tf2 \
-    ros-kinetic-tf2-ros \
-    ros-kinetic-visualization-msgs \
-    ros-kinetic-xacro \
-    ros-kinetic-rosbridge-server
+    qtbase5-dev && \
+    rm -rf /var/lib/apt/lists/*
+
+#
+# Scan Autoware packages to install dependencies
+#
+# Ignore ament dependencies for now
+RUN cd && mkdir -p /tmp/Autoware
+COPY ./ /tmp/Autoware
+RUN apt-get update && \
+        rosdep init && \
+        rosdep update && \
+        rosdep install -y --from-paths /tmp/Autoware/ros/src --ignore-src --rosdistro kinetic --skip-keys="ament_lint_common ament_lint_auto ament_cmake_gtest ament_cmake ament_cmake_lint_cmake" && \
+        rm -rf /var/lib/apt/lists/* && \
+        rm -rf /tmp/Autoware  # Removed for clean environment
+
+RUN echo "deb http://repo.ros2.org/ubuntu/main xenial main" | tee /etc/apt/sources.list.d/ros2-latest.list
+RUN apt-get update
+
+ARG ROS2_DISTRO
+RUN apt-get install -y --no-install-recommends \
+    ros-${ROS2_DISTRO}-ament-clang-format \
+    ros-${ROS2_DISTRO}-ament-cmake \
```

Comment:
Many packages (like ros-${ROS2_DISTRO}-ament-cmake) depend on many of the packages below. Even with --no-install-recommends, it should still automatically install the dependencies. Is it required for some reason to list them out below if they are dependencies of another package? ament-cmake covers -core, -export-definitions, -export-dependencies, -export-include-directories, -export-libraries, -export-link-flags, -libraries, -python, -target-dependencies, -test, and -workspace.

Reply:
@JWhitleyAStuff I think it is very useful for clarity of what we do here. These packages (and ROS2 in parallel with ROS1) are just a temporary solution to facilitate the transition to ROS2 and will go away once ROS_DISTRO becomes crystal. However at the moment it is nice to have it here listed in the most explicit way.